warehouse/utils/wsgi.py

```diff
@@ -66,11 +68,24 @@ def __call__(self, environ, start_response):
             }
 
             host = environ.get("HTTP_WAREHOUSE_HOST", "")
+
         # If we're not getting headers from a trusted third party via the
         # specialized Warehouse-* headers, then we'll fall back to looking at
         # X-Forwarded-* headers, assuming that whatever we have in front of us
         # will strip invalid ones.
         else:
+            # If there IS a token, but it doesn't match, then tell us about it.
```

Comment:
You probably don't need the compare here, there's no possible way for the compare digest to return `True`.